src/pages/Sidebar/HappinessForm.tsx

```diff
@@ -0,0 +1,213 @@
+import { validateHappiness, formatDate, formatHappinessNum } from "../../utils";
+import TextareaAutosize from "react-textarea-autosize";
+import HappinessNumber from "../../components/HappinessNumber";
+
+export default function HappinessForm({ height }: { height: number }) {
+  const { api } = useApi();
+
+  const [comment, setComment] = useState("");
+  const [submissionStatus, setSubmissionStatus] = useState("Updated");
+  // TODO refactor to use useRef! the let def was being reassigned
+  // on every recomposition
+  const postHappinessTimeout = useRef<number | undefined>(undefined);
+  const isInitialRender = useRef(true);
+
+  const [radioValue, setRadioValue] = useState(2);
+  const [selDate, setSelDate] = useState(new Date());
+  const [happiness, setHappiness] = useState(-1);
+
+  const UNSUBMITTED = "Unsubmitted (enter number to submit)";
+  const UPDATING = "Updating...";
+  const UPDATED = "Updated";
+  const ERROR = "Error loading/retrieving happiness";
+
+  const postHappinessMutation = useMutation((newHappiness: NewHappiness) =>
+    api.post("/happiness/", newHappiness),
+  );
+
+  useEffect(() => {
+    if (happiness === -1) {
+      setSubmissionStatus(UNSUBMITTED);
+    } else {
+      setSubmissionStatus(UPDATING);
+      clearTimeout(postHappinessTimeout.current);
+      postHappinessTimeout.current = setTimeout(() => {
+        postHappinessMutation.mutate({
+          value: happiness,
+          comment: comment,
+          timestamp: formatDate(selDate),
+        });
+      }, 1000);
+    }
+  }, [comment, happiness]);
+
+  useEffect(() => {
+    if (postHappinessMutation.isSuccess && happiness !== -1) {
+      setSubmissionStatus("Updated");
+    }
+    refetch();
+  }, [postHappinessMutation.isSuccess]);
+
+  useEffect(() => {
+    setSelDate(() => {
+      if (radioValue === 1) {
+        return new Date(
+          new Date().getFullYear(),
+          new Date().getMonth(),
+          new Date().getDate() - 1,
+        );
+      } else {
+        return new Date();
+      }
+    });
+  }, [radioValue]);
+
+  // autofill old entry on load
+  const {
+    isLoading,
+    data,
+    isError,
+    refetch,
+  }: {
+    isLoading: boolean;
+    data: Happiness[] | undefined;
+    isError: boolean;
+    refetch: (
+      queryFnArgs?: undefined,
+    ) => Promise<Happiness[] | undefined | unknown>;
+  } = useQuery(`happiness for user`, () => {
+    return api
+      .get("/happiness/", {
+        start: formatDate(
+          new Date(
+            new Date().getFullYear(),
+            new Date().getMonth(),
+            new Date().getDate() - 1,
+          ),
+        ),
+        end: formatDate(
+          new Date(
+            new Date().getFullYear(),
+            new Date().getMonth(),
+            new Date().getDate(),
+          ),
+        ),
+      })
+      .then((res) => res.data);
+  });
+
+  // react to initial query
+  useEffect(() => {
```

Comment:
Remove the remaining comments inside of this `useEffect`. You can keep the todo from design I guess actually.